File: ontrack-extension-svn/src/main/java/net/nemerosa/ontrack/extension/svn/model/SVNRevisionInfo.java
package net.nemerosa.ontrack.extension.svn.model;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class SVNRevisionInfo {

    private final long revision;
    private final String author;
    private final LocalDateTime dateTime;
    private final String path;
    private final String message;
    private final String revisionUrl;

    public SVNLocation toLocation() {
        return new SVNLocation(path, revision);
    }
}

File: ontrack-extension-svn/src/main/java/net/nemerosa/ontrack/extension/svn/model/SVNRevisionInfo.kt
package net.nemerosa.ontrack.extension.svn.model

import java.time.LocalDateTime

class SVNRevisionInfo(
        val revision: Long,
        val author: String,
        val dateTime: LocalDateTime,
        val path: String,
        val message: String,
        val revisionUrl: String
) {
    fun toLocation() = SVNLocation(path, revision)
}
